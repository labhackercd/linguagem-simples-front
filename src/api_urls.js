module.exports = Object.freeze({

    //NAME CONSTANTS
    APPLICATION_NAME: "Estudio",

    //APPLICATION URL'S
        APPLICATION_FRONT_BASE_URL: "www.teste.com",  //APPLICATION FRONT BASE URL
        //APPLICATION_SERVER_API_BASE_URL: 'http://localhost:8000/api/', //APPLICATION API SERVER BASE URL
        APPLICATION_SERVER_API_BASE_URL: 'https://acompanhe.camara.leg.br/server/api/',
        APPLICATION_RESET_PASSWORD_URL: 'https://acompanhe.camara.leg.br/server/password_reset/',

    //ROUTES
        INITIAL_PAGE_URL: '/', //Application Login and Initial page
        ESTUDIO_PAGE_URL: '/estudio', //Application Estudio Page
        DASHBOARD_PAGE_URL: '/dashboard/:dashboardId', //Application Dashboard page
        DASHBOARD_BASE_URL: '/dashboard/',

    //API'S CAMARA DOS DEPUTADOS
        API_CD_EVENTOS_PLENARIO: 'https://dadosabertos.camara.leg.br/api/v2/eventos?codTipoEvento=110',// 110 code corresponds to Plenary
        API_CD_EVENTOS: 'https://dadosabertos.camara.leg.br/api/v2/eventos/',
        API_CD_ORIENTACAO_VOTACAO: 'https://liderancadigital.camara.leg.br/proxy_pass/secod-wsVotDecom-orientacao',
        API_CD_ITENS_EM_VOTACAO: 'https://infoleg.camara.gov.br/wsVotDecom/votacao/itens-em-votacao-na-reuniao/',

    //API'S BACKEND
        //AUTH URL'S
            TOKEN_REFRESH_URL: '/token/refresh/', //Application Api Token REFRESH PATH
            TOKEN_OBTAIN_URL: '/token/obtain/', //Application Api Token PATH
            TOKEN_VERIFY_URL: '/token/verify/', //Application Api Token Verify

        //FILE-VIDEO
            API_FILE_VIDEO_URL:'/file-video/', //Application end-point to obtain the source of specific video

        //NEWS
            API_NEWS_URL:'/newsâ€‹/news_list', //Obtain news list
            API_NEWS_SEARCH_URL:'/news-search/', //Obtain specific news using parameters

        //PUBLICATIONS
            API_PUBLICATIONS_URL:'/publications/',

        //RADIO AGENCY
            API_RADIO_AGENCY_URL:'/radioagency/',
            API_RADIO_AGENCY_SEARCH_URL:'/radioagency-search/',

        //RADIO CAMARA
            API_RADIO_CAMARA_URL:'/radiocamara/',
            API_RADIO_CAMARA_SEARCH_URL:'/radiocamara-search/',

        //SAVED CONTENTS
            API_SAVED_CONTENTS_URL:'/saved-contents/',

        //SESSIONS
            API_SESSIONS_URL:'/sessions/',

        //TV CAMARA
            API_TV_CAMARA_URL:'/tvcamara/',
            API_TV_CAMARA_SEARCH_URL:'/tvcamara-search/',

        //VIDEOS
            API_VIDEOS_SNIPPETS:'/videos-session/'
  });
