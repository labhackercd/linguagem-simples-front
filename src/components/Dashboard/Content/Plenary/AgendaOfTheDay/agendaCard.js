import React from 'react';
import { Grid, Typography } from '@material-ui/core';
import IconButton from '@material-ui/core/IconButton';
import Box from '@material-ui/core/Box';
import Paper from '@material-ui/core/Paper'

import LaunchIcon from '@material-ui/icons/Launch';
import Divider from '@material-ui/core/Divider';



export default function AgendaCard(props){
    return (
        <Box width="97%" >
            <Paper elevation={0} >
                        <Box m={1}>
                            <Grid container>
                                <Grid item xs={10}>
                                    <Box fontWeight="fontWeightRegular">
                                        <a rel={'external noopener noreferrer'} target="_blank" href={"https://www.globo.com"} style={{textDecoration: "none"}}>
                                            <Typography variant="h6" style={{ color: "#007E5A" }}>
                                                Titulo
                                            </Typography>
                                        </a>
                                    </Box>
                                </Grid>
                                <Grid item xs={2}>
                                    <Box display="flex" justifyContent="flex-end">
                                        <IconButton aria-label="Ir para Twitter" size="small">
                                            <LaunchIcon  fontSize="inherit" />
                                        </IconButton>     
                                    </Box>
                                </Grid>
                                <Grid item xs={12}>
                                  <Box fontSize={11}>
                                    <Typography style={{ color: "gray" }}>
                                         texto descrição texto descrição  texto descrição  texto descrição  texto descrição  texto descrição texto descrição  texto descrição  texto descrição 
                                    </Typography>
                                    </Box>
                                </Grid>
                            </Grid>
                        </Box>
                    <Divider variant="middle" />
            </Paper>
        </Box>
    );
}