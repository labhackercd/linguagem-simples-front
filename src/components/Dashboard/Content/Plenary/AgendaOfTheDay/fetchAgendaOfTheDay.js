import axios from 'axios'
import {API_CD_PAUTA_SESSAO_PLENARIO} from '../../../../../api_urls'

export default async function fetchAgendaOfTheDay(sessionIdDadosAbertos) {

    const url = API_CD_PAUTA_SESSAO_PLENARIO+sessionIdDadosAbertos+"/1"
    
    const response= await axios.get(url)
    //console.log(response)
    return  response.data;
}
