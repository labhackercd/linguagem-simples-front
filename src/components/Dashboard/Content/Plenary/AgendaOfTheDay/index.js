import React from 'react';

import { Grid} from '@material-ui/core';
import Box from '@material-ui/core/Box';
import List from '@material-ui/core/List'
import CircularProgress from '@material-ui/core/CircularProgress';

import AgendaCard from './agendaCard'
import fetchAgendaOfTheDay from './fetchAgendaOfTheDay'


export default class AgendaOfTheDayComponent extends React.Component {
    
  constructor(props){
    super(props);
    this.state = { 
        pautas: null, 
        dataLoaded: false,
        sessionIdDadosAbertos: this.props.sessionIdDadosAbertos
    };
  }

  fetchAgenda = async term => {
    var data = null
      if(this.state.sessionIdDadosAbertos !== undefined){
        data = await fetchAgendaOfTheDay(this.state.sessionIdDadosAbertos);
        this.setState({pautas:data.dados})
        this.setState({dataLoaded:true});
      }else{
        //Nothing to do
      }
  };

  componentDidMount(){
      this._isMounted = true;

      if(this._isMounted){
          this.fetchAgenda();
      }
  }

  render(){

    if(!this.state.dataLoaded){
      return (
        <Box display="flex" justifyContent="center" alignItems="center" minWidth="40vh" minHeight="10vh" width={1}>
          <CircularProgress></CircularProgress>
        </Box>
      )

    }

    return (
      <Box width={'99%'} height={"85%"} style={{ overflow: 'auto'}}>
        <Grid container>
          <Grid item xs={12}>
              <List style={{maxHeight: '260px', maxWidth:'98%', overflow: 'auto'}}>            
                {this.state.pautas.map((pauta) => (
                    <li key={`section-${pauta.proposicao_.id}`}>
                        <Box my={0.5}><AgendaCard pautaInfo={pauta} sessionIdDadosAbertos={this.state.sessionIdDadosAbertos}></AgendaCard></Box>
                    </li>
                ))}
              </List>
          </Grid>
        </Grid>
      </Box>
    )
  }
}
