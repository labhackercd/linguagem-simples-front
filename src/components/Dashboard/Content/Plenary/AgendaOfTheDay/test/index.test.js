import React from 'react';
import { mount} from 'enzyme';
import AgendaOfTheDay from './../index'
import axios from 'axios'
import MockAdapter from "axios-mock-adapter"
import {API_CD_PAUTA_SESSAO_PLENARIO} from '../../../../../../api_urls'



describe('Testing lifeclycle ', () => {

  var mockInstance = new MockAdapter(axios);
  const mockedResponse = {
    "data": [
      {
        "ordem": 1,
        "proposicaoNome": "REQ 2144/2020",
        "codProposicao": 2260444,
        "situacaoItem": 2,
        "codParlamentarRelator": null,
        "nomRelator": null,
        "partidoRelator": null,
        "ufRelator": null,
        "bolRelatorPresente": 0
      },
      {
        "ordem": 2,
        "proposicaoNome": "MPV 961/2020",
        "codProposicao": 2251926,
        "situacaoItem": 2,
        "codParlamentarRelator": 1117208,
        "nomRelator": "JoÃ£o Campos",
        "partidoRelator": "REPUBLIC  ",
        "ufRelator": "GO",
        "bolRelatorPresente": 1
      },
    ],
    "status": 200,
    "statusText": "OK",
    "headers": {
      "cache-control": "private",
      "content-length": "406",
      "content-type": "application/json;charset=utf-8"
    },
    "config": {
      "url": "https://liderancadigital.camara.leg.br/proxy_pass/ws-pauta-itens-da-pauta/59896/1",
      "method": "get",
      "headers": {
        "Accept": "application/json, text/plain, */*"
      },
      "transformRequest": [
        null
      ],
      "transformResponse": [
        null
      ],
      "timeout": 0,
      "xsrfCookieName": "XSRF-TOKEN",
      "xsrfHeaderName": "X-XSRF-TOKEN",
      "maxContentLength": -1
    },
    "request": {}
  }
  
  test("should render the AgendaOfTheDay component and match snapshot ", async (done) => {

    const sessionIdDadosAbertos = 1;
  
    await mockInstance.onGet(API_CD_PAUTA_SESSAO_PLENARIO+sessionIdDadosAbertos+"/1").reply(200,mockedResponse.data)
            
    const component = mount(<AgendaOfTheDay sessionIdDadosAbertos={1}/>);
  
    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
  
    done();
  });

  

  afterAll(() => {
      mockInstance.restore();
  });


});