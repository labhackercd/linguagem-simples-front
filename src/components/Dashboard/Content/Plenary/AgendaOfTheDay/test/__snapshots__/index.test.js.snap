// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing lifeclycle  should render the AgendaOfTheDay component and match snapshot  1`] = `
<AgendaOfTheDayComponent
  sessionIdDadosAbertos={1}
>
  <Styled(MuiBox)
    alignItems="center"
    display="flex"
    justifyContent="center"
    minHeight="10vh"
    minWidth="40vh"
    width={1}
  >
    <div
      className="MuiBox-root MuiBox-root-1"
    >
      <WithStyles(ForwardRef(CircularProgress))>
        <ForwardRef(CircularProgress)
          classes={
            Object {
              "circle": "MuiCircularProgress-circle",
              "circleDisableShrink": "MuiCircularProgress-circleDisableShrink",
              "circleIndeterminate": "MuiCircularProgress-circleIndeterminate",
              "circleStatic": "MuiCircularProgress-circleStatic",
              "colorPrimary": "MuiCircularProgress-colorPrimary",
              "colorSecondary": "MuiCircularProgress-colorSecondary",
              "indeterminate": "MuiCircularProgress-indeterminate",
              "root": "MuiCircularProgress-root",
              "static": "MuiCircularProgress-static",
              "svg": "MuiCircularProgress-svg",
            }
          }
        >
          <div
            className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style={
              Object {
                "height": 40,
                "width": 40,
              }
            }
          >
            <svg
              className="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx={44}
                cy={44}
                fill="none"
                r={20.2}
                strokeWidth={3.6}
                style={Object {}}
              />
            </svg>
          </div>
        </ForwardRef(CircularProgress)>
      </WithStyles(ForwardRef(CircularProgress))>
    </div>
  </Styled(MuiBox)>
</AgendaOfTheDayComponent>
`;
